docs/source/observing.rst

```diff
@@ -0,0 +1,67 @@
+Observing
+=========
+
+This package includes some utilities for observing the progress and performance of running pipelines. In a similar
+manner to the python logging framework, observers are meant to be used as a global facility whose handling can be
```

Comment:
It might be better to overview the observer's role and mechanism in one or two sentences for more straightforward understanding

Reply:
I put the paragraph just below this one which I hope drives the point of what they are for